pkg/pbq/store/interfaces.go

```diff
@@ -32,3 +35,11 @@ type Store interface {
 	// GC does garbage collection and deletes all the messages that are persisted
 	GC() error
 }
+
+// StoreProvider defines the functions for store implementation
+type StoreProvider interface {
+	// CreatStore returns a new store instance.
+	CreatStore(context.Context, partition.ID) (Store, error)
```

Comment:
intentional spelling mistake to follow `creat` syscall? :)

Reply:
;)